csharp/src/Google.Protobuf.Test/Reflection/CustomOptionsTest.cs

```diff
@@ -75,107 +76,128 @@ OptionFetcher<E> EnumFetcher<E>(CustomOptions options)
         public void ScalarOptions()
```

Comment:
would it make more sense to have a separate test case ScalarOptions_OldGeneratedCode() that would test the legacy code? Such test could also have a comment that explains what it is testing (and perhaps link the github issue).